Shopify/cli
# [Feature] Add locales to 'customer-accounts-ui' extensions #43321

✅ Add `locales` folder to `customer-accounts-ui`
✅ Upload `locales` when deploying a `customer-accounts-ui` extension

<br>
Big thanks goes to @TobiasBales!
<br>
<br>

### WHY are these changes introduced?

We want to provide future engineers an easy and intuitive way to add localization into their `customer-accounts-ui` extensions. In addition we also want a streamlined process, so ALL future `customer-accounts-ui` extensions follow the same translation pattern.

**Ticket:** 
https://github.com/Shopify/core-issues/issues/43321

**⚠️ Important:** 
This PR can only be merged after https://github.com/Shopify/shopify/pull/372949 has been merged into Shopify core.

### WHAT is this pull request doing?

This PR adds `locales` to `customer-accounts-ui` extensions. With this PR, all `customer-accounts-ui` extensions which have been generated via the cli `yarn scaffold (customer-accounts-ui)` will include a `locales` folder with a default `en.default.json` and `fr.json` translation file.


### Measuring impact

How do we know this change was effective? Please choose one:


### How to test your changes?

In order to test those changes, please execute the following steps:
1. clone repo
2. checkout this branch
3. `bin/spin yarn create-app --path ..`
4. `yarn shopify app scaffold extension --path ../your-app-name` (please select `Customer Account UI` when asked by the CLI.
5. inside your newly created app, you should find a locales folder: `your-app-name/extensions/your-extension-name/locales`
